### PR TITLE
Task/WG-383: add queue for heavy tasks

### PR DESCRIPTION
**Pull request by nathanfranklin**
## Overview: ##

Add `heavy` queue for computationally intensive tasks. In this PR, the queue will be used by:
* Point cloud processing, which takes a lot of CPU and memory when processing large point clouds.  
* The street view uploading task can take a long time (even though it doesn't use a lot of CPU or memory).

The number of workers for both queues is:
*  `default` queue uses Celery's default behavior: nproc number of worker processes.
* `heavy` queue is set to nproc/2 number of workers (so intensive/long tasks don't monopolize things)
 
Note: While we've discussed categorizing tasks based on their true intensity/need (e.g., differentiating between small and large point cloud processing), this PR doesn't implement that. Instead, it makes incremental improvements by just having a smaller queue for large tasks to i) reduce the risk of memory exhaustion during large point cloud processing, and ii) prevent system monopolization by intensive tasks.

## Related Jira tickets: ##

* [WG-383](https://tacc-main.atlassian.net/browse/WG-383)

## Testing Steps: ##
1. Run and ensure you load features like images and then also point clouds to ensure both queues are working.

### CODE REVIEW

**Review by rstijerina — APPROVED** [review]
LGTM

File: devops/geoapi-workers/docker-compose.yml [hunk 1]
```diff
@@ -18,7 +18,12 @@ services:
       driver: syslog
       options:
         tag: geoapi_workers
-    command: "celery -A geoapi.celery_app worker -l info"
+    command:  >
+      sh -c '
+      celery -A geoapi.celery_app worker -l info -Q default -n default_worker@geoapi &
+      celery -A geoapi.celery_app worker -l info -Q heavy --concurrency=6 -n heavy_worker@geoapi &
+      wait
```
Comment (rstijerina) [inline devops/geoapi-workers/docker-compose.yml:25]: Just curious - is this preferable over letting the process hang on the `celery` command?

File: geoapi/tasks/external_data.py [hunk 2]
```diff
@@ -167,7 +167,7 @@ def _handle_point_cloud_conversion_error(pointCloudId, userId, files, error_desc
                                     f"Processing failed for point cloud ({pointCloudId})!")
 
 
-@app.task(rate_limit="1/s")
+@app.task(queue='heavy')
```
Comment (sophia-massie) [inline geoapi/tasks/external_data.py:170]: Just a suggestion - Should we add a logger to let us know which queue a task is using? Probably not needed just curious if that would help us down the line.

**Review by sophia-massie — APPROVED** [review]

File: devops/geoapi-workers/docker-compose.yml (hunk 1 above)
Comment (nathanfranklin) [reply devops/geoapi-workers/docker-compose.yml:25]: @rstijerina  as we have two celery tasks, at least the first one needed to be ran in the background, `&` but second one could have been run in the foreground and then the wait skipped.
i didn't see a way to run celery and say, i want X concurrency on this queue and Y concurrency on this other queue

File: geoapi/tasks/external_data.py (hunk 2 above)
Comment (nathanfranklin) [reply geoapi/tasks/external_data.py:170]: the task itself gets logged but not which queue it was in. we could have a base celery task that our tasks use and could log that info. good idea when we make improvements to this.

**Pull request merged**